### ISSUE TITLE
Keras batch norm performance issue with parameter server strategy

### ISSUE THREAD

**Issue by bilisun** [tensorflow/tensorflow#43638]
**System information**

Have I written custom code (as opposed to using a stock
example script provided in TensorFlow): Yes
OS Platform and Distribution (e.g.,
Linux Ubuntu 16.04): Ubuntu 16.04
Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if
the issue happens on mobile device:
TensorFlow installed from (source or
binary): source
TensorFlow version (use command below): 1.14
Python version: - Bazel
version (if compiling from source):
GCC/Compiler version (if compiling from
source):
CUDA/cuDNN version: Cuda 10.1 cuDNN 7.6.5
GPU model and memory: V100

**Describe the current behavior**

Currently I'm observing a performance issue regarding Keras batch norm layer on tensorflow 1.14 with the following setup: Parameter server for multi-workers, mirrored strategy within each individual worker (8 GPUs per worker). All variables are cached on the worker cpus via `caching_device` to reduce parameter server <-> gpu communication.

Specifically, I'm wondering why https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/layers/normalization.py#L492 exists (first added in https://github.com/tensorflow/tensorflow/commit/88ad9949ef4ea6e07105a326a1d21c108cb2883a). This places all moving mean / variance update tensors, for each GPU, on the parameter server, causing communication between worker GPUs and the parameter server that does not scale (passing the tensor 64 times in my setup rather than 8 times in a 1 GPU per worker, 8 workers setup). Upon removing the colocation, the update tensors are placed on individual GPUs, and the communication disappears and distributed training with the parameter server scales properly. Does the colocation exist to properly synchronize the moving mean / variance updates?

**Describe the expected behavior**

I'd expect there to be no communication between individual GPUs and the parameter server, and the moving mean and variance update tensors placed on devices in a way that is efficient.

**Comment by amahendrakar** [comment]
@bilisun,
TensorFlow 1.x is not actively supported. Could you please update TensorFlow to v2.3 and check if you are facing the same issue. 

Also, in order to expedite the trouble-shooting process, could you please provide a minimal working example to reproduce the issue reported here. Thanks!

**Comment by bilisun** [comment]
Trying v2.3 might take some time, will check back in with you.

In the meantime, where can I find the rationale for https://github.com/tensorflow/tensorflow/commit/88ad9949ef4ea6e07105a326a1d21c108cb2883a#diff-c2a77473d7da4416da00fd1c7050f26dR367? Thank you!

**Comment by amahendrakar** [comment]
> In the meantime, where can I find the rationale for [88ad994#diff-c2a77473d7da4416da00fd1c7050f26dR367](https://github.com/tensorflow/tensorflow/commit/88ad9949ef4ea6e07105a326a1d21c108cb2883a#diff-c2a77473d7da4416da00fd1c7050f26dR367)?

@bilisun,
Could you please check the associated PR for this.

> Trying v2.3 might take some time, will check back in with you.

Also, any updates regarding this? Is this still an issue? Thanks!

**Comment by bilisun** [comment]
@amahendrakar There isn't really an associated PR, the PR adding that change seems like a generic cherry-pick of many, many commits (https://github.com/tensorflow/tensorflow/pull/20064). Trying v2.3 won't be feasible for a long time for me.
